### PR TITLE
[NEXTGEN-18139]- Incorrect syntax/usage of nz_azConnector command run should return error

### PR DESCRIPTION
**Pull request by ram3425**
JIRA Story (/Task) link: https://jsw.ibm.com/browse/NEXTGEN-18139

Problem: Incorrect syntax/usage of nz_azConnector command run should return error

Solution: Checking for all the flags of nz_azConnector, whether '-' is passed or not. If not passed, then erroring out. 

Testing: Checked with all the possible solutions. Here are the outputs for some cases.

1) If '-' is not passed in storage-account then it is erroring out a message. Same for other flags as well.
```
[nz@beziques1 nz_azConnector]$  go run nz_azConnector.go -db=TESTDB -dir=/home/nz -npshost=xxxx storage-account=backupandrestoreconcerto -key=xxxx -container=batchop -upload -uniqueid=ram -blocksize=2 -logfiledir=/tmp
Error : Incorrect syntax. Missing '-' before command line argument:  storage-account=backupandrestoreconcerto
exit status 1
[nz@beziques1 nz_azConnector]$
```

2) If '-' is passed for all the flags then backup is successful. 
```
[nz@beziques1 nz_azConnector]$  go run nz_azConnector.go -db=TESTDB -dir=/home/nz -npshost=xxxx -storage-account=backupandrestoreconcerto -key=xxxx -container=batchop -upload -uniqueid=ram -blocksize=2 -logfiledir=/tmp
2024-03-01 09:20:52 EST  [INFO] Azure account name : backupandrestoreconcerto
2024-03-01 09:20:52 EST  [INFO] Azure container : batchop
2024-03-01 09:20:52 EST  [INFO] Number of blocks to upload/download in parallel : 16
2024-03-01 09:20:52 EST  [INFO] Block size in MB to upload/download file 2
2024-03-01 09:20:52 EST  [INFO] Backup/Restore directory : [/home/nz]
2024-03-01 09:20:52 EST  [INFO] DB name : TESTDB
2024-03-01 09:20:52 EST  [INFO] Nps hostname : beziques1.fyre.ibm.com
2024-03-01 09:20:52 EST  [INFO] BackupsetID : ALL
2024-03-01 09:20:52 EST  [INFO] UniqueID : ram
2024-03-01 09:20:52 EST  [INFO] Number of files to upload/download in parallel : 6
2024-03-01 09:20:52 EST  [INFO] Uploading backup data to azure cloud from backup dir /home/nz
2024-03-01 09:20:52 EST  [INFO] Uploading file : /home/nz/Netezza/beziques1.fyre.ibm.com/TESTDB/20231212085307/1/FULL/md/loc1/locations.txt
2024-03-01 09:20:52 EST  [INFO] Uploading file : /home/nz/Netezza/beziques1.fyre.ibm.com/TESTDB/20231212085307/1/FULL/md/schema.xml
2024-03-01 09:20:52 EST  [INFO] Upload successful. Total files uploaded: 2

```

### CODE REVIEW

**Comment by ram3425** [comment]
The code changes are from line number 387 to 394. As we need to format the code, I triggered "gofmt -w nz_azConnector.go" which shows changes in whole file.

**Comment by snehalunawat** [comment]
Changes looks good. Approved

**Comment by abhishekjog** [comment]
Please check flag.Visit and VisitAll.

**Comment by ram3425** [comment]
Hi @abhishekjog, by using this Visit() function as it is a language construct it will not read anything once it get no hyphen flag.
By VisitAll() we are reading all the flags and preparing that unset variable. And after that Visit() is removing the flags from the unset, which are set from the command.

`go run nz_azConnector.go -db=TESTDB -dir=/home/nz -npshost=xxxxx -storage-account=backupandrestoreconcerto -key=xxxxxxxxxx -container=batchop -upload -uniqueid=ram -blocksize=2 logfiledir=/tmp -backupset=yuu`

Here it can't remove backupset from the unset variable as it will not read anything once reached logfiledir. After logfiledir it will ignore all so it is not possible to delete backupset from unset. And there is no way to check whether '-' is passed or not using Visit() because it is a inbuilt functionality.

File: bnr-utils/nz_azConnector/nz_azConnector.go [hunk 1]
```diff
@@ -1,500 +1,508 @@
...
+	var backupinfo BackupInfo
+	var othargs OtherArgs
+
+	// parse input args
+	parseArgs(&conn, &backupinfo, &othargs)
+	flag.Parse()
+
+	//Checking whether '-' is passed or not in flags
+	args := flag.Args()
+	for _, arg := range args {
```
Comment (abhishekjog) [inline bnr-utils/nz_azConnector/nz_azConnector.go]:
```suggestion
	if len(flag.Args()) != 0 {
                fmt.Println("Error : Incorrect syntax. Missing '-' before command line argument: ", flag.Args()[0])
                os.Exit(1)
         }
```

**Review by abhishekjog — CHANGES_REQUESTED** [review]
This is all we need.

File: bnr-utils/nz_azConnector/nz_azConnector.go (hunk 1 above)
Comment (ram3425) [reply bnr-utils/nz_azConnector/nz_azConnector.go]: Hi @abhishekjog, I made the required changes.

**Comment by ram3425** [comment]
The code changes are from line number 388 to 392.

File: bnr-utils/nz_azConnector/nz_azConnector.go [hunk 2]
```diff
@@ -1,500 +1,506 @@
...
+func main() {
+	var conn Conn
+	var backupinfo BackupInfo
+	var othargs OtherArgs
+
+	// parse input args
+	parseArgs(&conn, &backupinfo, &othargs)
+	flag.Parse()
+
+	//Checking whether '-' is passed or not in flags
```
Comment (abhishekjog) [inline bnr-utils/nz_azConnector/nz_azConnector.go]: Both these checks should be after MultiWriter initialisation. Errors should not be printed only to stdout, but also to log file.

**Review by abhishekjog — CHANGES_REQUESTED** [review]
please check the comment. we should not be logging anything to stdout alone. log/error should also go to log file.

File: bnr-utils/nz_azConnector/nz_azConnector.go (hunk 2 above)
Comment (ram3425) [reply bnr-utils/nz_azConnector/nz_azConnector.go]: Done with these changes.

**Comment by ram3425** [comment]
Code changes are from line number 401 to 410.

**Review by abhishekjog — APPROVED** [review]

**Pull request merged**